### ISSUE TITLE
Bringing back sculpting tools.

### ISSUE THREAD

**Issue by umiyuki** [umiyuki/Artstage-Issues#15]
Posted from trello.

Bringing back sculpting tools.
Make three basic modes: modeling, sculpting, texturing. It would also be nice if there could be a sculpted object and modeled object in the scene at the same time.
When this is done, it will be possible to implement tools for retopology - 4th mode.

**Comment by umiyuki** [comment]
Implementing sophisticated sculpting tools is very difficult.
Does it mean that you want to revive even with the previous version sculpt tool?

There are other great tools (Kodon, Medium) for sculpt, so it is better to export obj from there and work with Artstage.

In Artstage, there is also a workflow that create a mesh based on the rough sketch created with the sketch tool.

**Comment by Ekrons** [comment]
If it is difficult, then don't. I just thought that if the program was self-sufficient, users would switch less to other applications. Without adding sculpting to Artstage, users will switch to Kodon, get used to its interface, and, in addition, the developers plan to update the modeling tools, and that's it, users are unlikely to return to Artstage. Now Kodon is the best at sculpturing on the VR market.
It seems to me that if Artstage had at least some kind of sculpting, perhaps fewer people would switch to other applications. But it is not exactly.
No, if Kodon is Artstage friends, then fine)

**Comment by umiyuki** [comment]
This is a place to receive requests, not a place to threat developers.
You should also say to Blender developers:
"You must implement the VR editing feature. If you don't, the users will move to Artstage."
Why don't you don't that?
If so, do they respond to your request?
Generally speaking, such a behavior does not increase the motivation of the developer. It will be the opposite.

Do you want for such a boring chat feedback like this? Development resources are lost to write this.

If Kodon's update is infrequent, it is a terribly difficult task. Do you want me to do the work and reduce the frequency of updates?

**Comment by Ekrons** [comment]
umiyuki
Well, before that I thought you were a good developer, and even wrote a positive review about you and your Steam program.
https://steamcommunity.com/profiles/76561198931762468/recommended/523050/
I would like Codon to have the same interest as you. Or am I mistaken in you? Write.

With regards to Blender, they are not interested in adding the VR functions, as it is a small segment.
Your applications were originally created for VR, and it is foolish to create programs specializing in modeling, and not to develop or improve modeling tools.

Did I ask you to add in the application "calculator" or "currency converter"? No, I asked for something that would be useful for people like me, modelers. Since you have removed the possibility of sculpting, then your program is now focused on this, on modeling.
Those functions which I requested, the basis for creating complex models. If you do not need this, and you want to create only birdhouses in Artstage, please.

**Comment by umiyuki** [comment]
I do not intend to treat you as a child, but sometimes I wonder if you are writing a joke.
I feel that it's probably fun and I just reply.
In fact I am thankful for your enthusiasm.

The hypothesis that you say that Artstage users are just me (and you) is interesting. I think it's different, but let's say it is.
If so, do I have to fully comply with your request? I do not think so.
I appreciate your purchase of the Artstage, which is $ 20. Or you may have spent just for the support cost here. It may have been different if you received $ 300 a month like Maya.
Thank you for your timely investment. However, I have invested more than two years in the development of this app. Then so, Should I prioritize my own requirements?

I do not want to ignore your request.
However, I should avoid ruining the app, blindly to the demands of loud users.
Therefore I expected the voting function. However, it seems to take time until it is used by many users.
That would be a problem that time solves. You're in a hurry, but I am not.

Would you like to add a negative review on Artstage this way?
"Developers ignore users because they didn't add sculpting features"
I do not know whether it is always useful for the person who saw the review.

**Comment by Ekrons** [comment]
umiyuki
Okay, then good luck.

**Comment by Aitolda** [comment]
I think what you're trying to do for traditional mesh editing techniques in VR is fantastic, and one of your biggest strengths.  So many others are already doing volumetric sculpting so I appreciate you filling this other void.  I feel my $20 was well spent, considering that in the short time I've used this product I've witnessed your speedy response. Thanks for all the hard work!

**Comment by Ekrons** [comment]
I think the best indicator of gratitude and progress for the developer should be: the money invested by the user in the product and the time spent in this product. Aitolda, how much time did you spend in the program, that write such high words? According to Steam, it is 3 hours. Or I'm wrong?
![Image 2019, 05, 04, 21, 46, 50](https://user-images.githubusercontent.com/50044554/57183501-a4d47100-6eb6-11e9-90dd-a6c9d0997750.png)

**Comment by Ekrons** [comment]
I think the best indicator of gratitude and progress for the developer should be: the number of users of the product, the money invested by the user in the product and the time spent by the user in this product. Aitolda, how much time did you spend in the program, that you can write such high words?
According to Steam, it is almost 4 hours. Or I'm wrong?
![Image 2019, 05, 04, 21, 46, 50](https://user-images.githubusercontent.com/50044554/57190778-03393800-6f27-11e9-9f15-7ee27c78c80c.png)

**Comment by MarkMironov** [comment]
It would be great if Artstage had the opportunity to sculpt, with the possibility of choosing what the density of the mesh would be.
Even one brush (add, subtract) would be enough if only there was an opportunity to sculpt.

**Comment by umiyuki** [comment]
I'm honestly not sure, so I'd like someone to explain it, but aren't Artstage users wanting polygon modeling?
For sculpting in VR, there are already great software such as Medium and Kodon.
Why do you want to use the lower quality sculpting function in Artstage without using those softwares?
Does it really increase the value of the application?
What specific work scenario will you need it for?
For example, do you want to create a character with polygon modeling and create a tree with sculpt?So then create a character in Artstage, create a tree in Medium. Is there a problem with this approach?
If you want sculpting at Artstage just because it's cumbersome to go back and forth between software, you may be ask to add rigging feature at Artstage after sculpting feature.
Even if you want every features, it is impossible due to development resource constraints.

**Comment by MarkMironov** [comment]
@umiyuki 
I planned to use sculpting as the ability to create a basic form.
By modeling, we can’t quickly sketch out its dimensions and shape. But sculpting is possible. In Artstage, we could edit the mesh with polygonal tools created by the sculpt brushes.

Also, elements such as clouds, smoke, hair, landscape, could be created much faster. Without bothering with exporting and importing from other programs, and without modeling it polygonally.

We could use the Medium or the Kodon, but Kodon has many bugs, and the Medium does not work in SteamVR. To do this, you must either have the Oculus, or install Revive.
That is, for the sake of one possibility of sculpting, we have to do a lot of actions. You will also need to buy and study another program (Medium or MasterpieceVR, for example) and then switch between them and ArtStage just for one function.

You have implemented "sketch" function. Is it really so hard to implement a sculpting brush?
All we need is:
- a brush (adding and subtracting) with the possibility of changing the shape (sphere, cube, pyramid).
The stroke should change thickness depending on the pressure on the trigger. With the ability to enable or disable this feature.
- adjustment of poly density.

**Comment by MarkMironov** [comment]
https://youtu.be/CkKjc_d-gEU
Here is what we need.
Only a more detailed mesh, and the absence of an external cube.

**Comment by umiyuki** [comment]
You are using Rift S, so you don't need to worry about those who only have Vive. Plus, Medium is easy to use, so the learning effort isn't great.
I can't convince unless you tell me a more specific reason. Do you really make clouds with Artstage's sculpt feature and print it 3D?
If you only want to sketch a shape and draft it, the sketch stroke tool is sufficient. There is also an Add vertices tool. Of course, sketch images are also available. I don't feel the need for sculpting.

Restoring the previous sculpt means that I have to continue to maintenance it. I don't want to do it. This will no better than now.
If sculpting is added to Artstage in future, it does not restore the previous one, but recreates a better one from scratch.

**Comment by umiyuki** [comment]
Paradoxically, the need for a sculpting feature doesn't seem to be urgent, if you can't switch to Medium because you are bother to do it.

**Comment by umiyuki** [comment]
Generally speaking, there are two ways to make an application implement a new feature that isn't enthusiastic for the developer and is large.

The first is to suggest supporting labor costs during development. For example, Blender gets development funds to support development costs. AMD is making ongoing donations to help Blender support the Vulkan API.
For Artstage, $ 300 per day may be sufficient. If it takes 100 days to implement, improve, and maintain the sculpt feature, it costs $ 30,000.

If such a method is not practical, the second method is to imagine the developer's position.
If a developer decides to support sculpting, he will carry the risk of spending 100 days on development that would otherwise have earned $ 30,000.
The worst case that a developer envisions is that they have struggled to implement the sculpting feature, but no one really uses it and has no value at all. “It's hard to use!” “If you use something like this, it's better to switch to Medium!” And I'm afraid that it will end with adding 100 issues to this board.
In other words, developers want seek guarantees against the risks they carry unilaterally.
If someone says "If implemented, someone might use it.", There is no guarantee. Need actions to convince not only words.
If I am a user who is eager to have a developer implement sculpting, then I must first take the lead in order to convince the developer.
For example, actually create a number of works by combining Artstage and Medium, and try posting a picture.
Record the production video and prove that "If Artstage has a sculpt, work time should have been reduced by 10%."
So far, developers may be persuaded that “it is certainly worth $ 30,000 to implement sculpt”.

**Comment by MarkMironov** [comment]
@umiyuki 
Over the past few days, I’ve been doing just that, trying to figure out if sculpting is needed at Artstage.
![Image 2019, 10, 26, 10, 57, 42](https://user-images.githubusercontent.com/55280047/67616475-f9133a80-f7e1-11e9-8c08-7e5f21fe4e80.png)

So far, everything is fine in Medium, except that it can be difficult to remove small deformations of surface. But this is most likely due to the fact that I switched to high resolution too early.

Also, I have not yet tried to transfer the model from Artstage and back.
As I understand it, it will be something like:
1. Export of the model to the Medium. With subdiv on.
2. Launching Medium.
3. Import of the model in the Medium.
4. Creating a layer for the hair (or something else), sculpting hair.
5. Export this layer.
6. Import of hair into Artstage.
7. Fitting their position to the head model.

Instead:
1. Select a sculpting brush and sculpt hair.

I just thought that one tool would not be a hindrance, especially since the sculpting functions were already in the early versions of Artstage.
Even if this tool does not work for someone perfectly, they will not lose modeling tools from this.

>"And I'm afraid that it will end with adding 100 issues to this board."
100 issues will appear due to the fact that people are interested in this feature, right?
If they do not need sculpting, they will not create anything.
And while there is no sculpting, you cannot know for sure whether users need it.
Therefore, I wrote that experimental sculpting can be done.

But I understand you and agree with you.
If such a brush as I described above is too expensive to do, then it’s not worth it. At least modeling functions have not yet been perfected.
I am for ensuring that you do not suffer losses, because you already made a lot of useful functions and I am grateful to you. Therefore, if you decide that sculpting is not needed, so be it.

**Comment by umiyuki** [comment]
I have a clear understanding of the features I want and work on development. That's because I can't implement it without a clear understanding.
But some users may not realize what they need true. Customers who come to the home center say “I want a drill,” but what he really need is a hole in the wall. It's just a belief that he need a drill to make a hole in the wall. Home center's clerk may be able to introduce other suitable means than drill if consulted to make a hole in the wall.

You may think that sculpting is necessary to make smoke and hair. But there is a possibility, that polygon editing will be enough if even a proportional editing feature is added.

> 100 issues will appear due to the fact that people are interested in this feature, right?

If the initial request is not what the user really wanted, it will be realized that it wasn't what he really wanted after it was implemented, If he think wrong idea that "If more features are added, it will be what I want'' he may make a lot of new requests (what he does'nt need true) and get out of control.

I repeat in this thread over and over, but I don't think that all work must be completed on Artstage. I created a model in Artstage, import the model into Blender and rigging it. This is a natural task. The same goes for Medium where sculpting is required. If you want to sculpt in Artstage, you should also ask Medium developers to implement polygon modeling.
In the first place, the idea that a little improvement over the previous Artstage sculpt would be a substitute for Medium is a complete misunderstanding. It is system on another level.

**Comment by umiyuki** [comment]
> 1.Select a sculpting brush and sculpt hair.

If you only make hair with Medium, you only need one round trip. Suppose you lose 10 minutes to import and export between Artstage and Medium. If Artstage has a sculpt, will it save 10 minutes? No. Medium had better sculpting than Artstage, so working with Medium was reduced by 20 minutes. You lose for 10 minutes.

**Comment by MarkMironov** [comment]
@umiyuki I don’t know how others, but as for myself, I’ll say that from what I asked you to implement, I use absolutely everything all the time.

**Comment by umiyuki** [comment]
Keep in mind that the sculpting feature has already been retired as I have decided to retire it. The reason is that we thought we should concentrate on the implementation of polygon modeling.
You can be likened to a difficult position like a lawyer trying to turn a conviction in the first trial over in second trials. However, no one submits evidence to prove the validity of the sculpting implementation, just repeated, give me a sculpt! Is it possible for such a lawyer to reverse the verdict?

I have said this 100 times, "Sculpt with Medium". I am developing Artstage because there is no other VR polygon modeling software that I like available. If I prefer sculpting, use Medium. There is no need to develop Artstage.
Use Medium if you like sculpting. If you like polygon modeling, use Artstage. But you want sculpt to Artstage! When I ask why I need to implement sculpting in Artstage, no one returns a logical answer.

**Comment by umiyuki** [comment]
I wanted to settle this thread.
I recorded a video to prove that the Artstage / Medium collaboration is no probrem.

https://youtu.be/Uv92yG5RDgc

I imported the Artstage model into Medium, sculpted the hair with Medium, exported the hair, and loaded it again with Artstage.
It took only 5 minutes! I didn't even need to take off the headset! ridiculous!

Those who claim that the journey between Artstage and Medium is very difficult is lying. This video is evidence.
Sculpt should be do in Medium. I would like to close this thread if there is no evidence that can be argued by others.

**Comment by umiyuki** [comment]
It's a shame that this thread ended with a boring line.
It seems as if it was a no wanted request from the beginning. If that was the case, you don't need to lengthen the thread meaninglessly.
I was always serious. Because if I made a mistake, there was a risk of throwing a huge amount of time in my life for features that no one wanted.
I wasted time for this thread! Please stop making requests that you don't serious want.
If your attitude is unreliable, you also need to consider your other requests carefully.

**Comment by MarkMironov** [comment]
@umiyuki 
As I wrote earlier, I use all the functions that I ask. All that I ask is considered and balanced.
But I can’t influence your desire to do something. If this is not interesting for you and you do not want to implement the function at all, why force you to?

If sculpting were implemented, I would use it.
But you are right, there are more important functions that need to be implemented.

I agreed to close the topic because I weigh the pros and cons, as well as your words about the complexity of the development. I came to the conclusion that one function can not be implemented, although there is a need for it.

But I just won’t be able to step back from the fact that I need features #82, #5 , #59 .
These are the most important features now, without which you experience some difficulties in Artstage. Here it’s not possible to use a external program, since they are needed much more often than sculpting.

You ask not to create issues. But I already realize that the finale is near. Features that I needed, I asked almost everything.
I looked at all the features that I requested, and did not find any that I could refuse, including sculpting. But you cannot realize sculpting, therefore I had to refuse it through force.

**Comment by umiyuki** [comment]
If there is no progress in the next two months, close the issue.
